src/Api/Service/RevisionSaver.php

```diff
@@ -93,6 +93,8 @@ public function save( Revision $revision, EditInfo $editInfo = null ) {
 
 		if( $editInfo->getBot() ) {
 			$params['bot'] = true;
+			// set maxlag for noninteractive requests
+			$params['maxlag'] = 5;
```

Comment:
So, this should now come from edit info?

Reply:
Yes putting it in WikibaseApi would make sense.

Essentially all parts of code that take an EditInfo object and construct a $params array before sending it to MediawikiApi object should look at the EditInfo and add the maxlag param if it is set.
Within this wikibase library im pretty sure rather than talking to MediawikiApi directly everything actually talks to the WikibaseApi object. :)